codewind-che-sidecar/deploy-pfe/pkg/kube/kube.go

```diff
@@ -0,0 +1,54 @@
+package kube
+
+import (
+	"os"
+
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetKubeClientConfig retrieves the Kubernetes client config from the cluster
+func GetKubeClientConfig() clientcmd.ClientConfig {
+	// Instantiate loader for kubeconfig file.
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
Noticed we get the kube config in main.go as well but in a slightly different way. Should one method be reused?

Reply:
Good question - It's confusing but it's actually two separate configs/types. The one in main.go is the kube config used for accessing Kubernetes resources of type `*rest.Config`. The one in kube.go is the client config for accessing the Kube API server (of type `*clientcmd.ClientConfig`)

I'll rename `kubeconfig` in kube.go to be `clientconfig`